boards/DIM/Src/App/App_SevenSegDisplays.c

```diff
@@ -37,74 +51,119 @@ void App_SevenSegDisplays_Destroy(struct SevenSegDisplays *const seven_seg_displ
     free(seven_seg_displays);
 }
 
-ExitCode App_SevenSegDisplays_SetHexDigits(
-    const struct SevenSegDisplays *const seven_seg_displays,
-    const uint8_t                        hex_digits[],
-    size_t                               num_hex_digits)
+ExitCode
+    App_SevenSegDisplays_SetHexDigits(const struct SevenSegDisplays *seven_seg_displays, const uint8_t hex_digits[])
 {
-    if (num_hex_digits > NUM_SEVEN_SEG_DISPLAYS || num_hex_digits == 0)
-    {
-        return EXIT_CODE_INVALID_ARGS;
-    }
+    size_t len_hex_digits = 9; // TODO: This is bad, should get from somewhere else
 
-    // If any of the input digits is invalid, we don't write anything to the
-    // 7-segment displays at all.
-    for (size_t i = 0; i < num_hex_digits; i++)
+    for (size_t i = 0; i < len_hex_digits; i++)
     {
         if (hex_digits[i] >= NUM_HEX_DIGITS)
         {
             return EXIT_CODE_INVALID_ARGS;
         }
     }
 
-    for (size_t i = 0; i < NUM_SEVEN_SEG_DISPLAYS; i++)
+    return EXIT_CODE_OK;
+}
+
+void App_Set_Digits(const struct SevenSegDisplays *seven_seg_displays, uint8_t index)
+{
+    for (int i = index; i < NUM_IN_GROUP + index; i++)
```

Comment:
Functions name should be `(App/Io)_(file name)_(function name)`
So `App_SevenSegDisplays_SetDigits` 

The name `index` is pretty vague, consider renaming to something like `group_index` so it's clear this refers to index of group, not index of digit or something else. Also could make a new enum of the 3 groups and pass that instead.

You should add some error checking too, if someone passed a 4 for ex. the program would crash.

Also, it doesn't look like function will be used in any other files? If so it should be removed from the header file and declared statically in this .c file.

Reply:
Also, it doesn't look like function will be used in any other files? If so it should be removed from the header file and declared statically in this .c file.


Naming convention fixed, 

Not really sure if I understand what you mean by passing a 4, I don't think it'll crash, just edit the incorrect places in the digit array. The error checking is done before this function is called for incorrect numbers, as was the fashion of the old code so I stuck with it believing it was convention (?)